pkg/oidc/token_request.go

```diff
@@ -40,6 +40,29 @@ var AllGrantTypes = []GrantType{
 
 type GrantType string
 
+const (
+	AccessTokenType  TokenType = "urn:ietf:params:oauth:token-type:access_token"
+	RefreshTokenType TokenType = "urn:ietf:params:oauth:token-type:refresh_token"
+	IDTokenType      TokenType = "urn:ietf:params:oauth:token-type:id_token"
+	JWTTokenType     TokenType = "urn:ietf:params:oauth:token-type:jwt"
+)
+
+var AllTokenTypes = []TokenType{
+	AccessTokenType, RefreshTokenType, IDTokenType, JWTTokenType,
+}
+
+type TokenType string
+
+func (t TokenType) IsValid() bool {
```

Comment:
Here `IsValid()` is in context of the framework itself, not RFC implementation

> Other URIs MAY be used to indicate other token types.
Yes RFC allows other token types, but their support is not implemented in this PR. We can't just allow any token type here since it won't be handled by the framework. It is applied limitation that can be dropped in the future with custom token types support implementation

Reply:
Please do change it to `IsSupported()`